test/e2e/specs/editor/various/invalid-block.spec.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Invalid blocks', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should show an invalid block message with clickable options', async ( {
+		editor,
+		page,
+	} ) => {
+		// Create an empty paragraph with the focus in the block.
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( 'hello' );
+
+		// Change to HTML mode and close the options.
+		await editor.clickBlockOptionsMenuItem( 'Edit as HTML' );
+
+		// Focus on the textarea and enter an invalid paragraph.
+		await editor.canvas
+			.getByRole( 'document', { name: 'Block: Paragraph' } )
+			.getByRole( 'textbox' )
+			.fill( '<p>invalid paragraph' );
+
+		// Takes the focus away from the block so the invalid warning is triggered.
+		await editor.saveDraft();
+
+		// Click on the 'three-dots' menu toggle.
+		await editor.canvas
+			.getByRole( 'document', { name: 'Block: Paragraph' } )
+			.getByRole( 'button', { name: 'More options' } )
+			.click();
+
+		await page
+			.getByRole( 'menu', { name: 'More options' } )
+			.getByRole( 'menuitem', { name: 'Resolve' } )
+			.click();
+
+		// Check we get the resolve modal with the appropriate contents.
+		await expect(
+			page
+				.getByRole( 'dialog', { name: 'Resolve Block' } )
+				.locator( '.block-editor-block-compare__html' )
```

Comment:
I guess we can just omit this class name selector as this doesn't provide any helpful accessible query?

Reply:
This narrows down the text content we want to assert. The dialog contains other text.

![CleanShot 2024-01-04 at 13 15 53](https://github.com/WordPress/gutenberg/assets/240569/4108c656-bc48-40d1-a568-77d34f343cc3)

Reply:
Same problem with `toContainText`. Considering that tests assert resolution diff, I think it's okay to use class name locator.